dashboard/app/models/pd/workshop.rb

```diff
@@ -192,7 +192,8 @@ def self.in_state(state)
       when STATE_ENDED
         where.not(started_at: nil).where.not(ended_at: nil)
       else
-        raise "Unrecognized state: #{state}"
+        raise "Unrecognized state: #{state}" if error_on_bad_state
+        none
```

Comment:
This is a query argument, not the workshop state. The workshop itself is always in a valid [state](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/workshop.rb#L263-L267)